Assets/Scripts/Game/UserInterface/SpellIconCollection.cs

```diff
@@ -228,42 +230,67 @@ void LoadSpellIconPacks()
             // Start with all the atlases in the spell icons streaming assets path
             string sourcePath = Path.Combine(Application.streamingAssetsPath, sourceFolderName);
             string[] atlasPaths = Directory.GetFiles(sourcePath, "*.png");
-            if (atlasPaths == null || atlasPaths.Length == 0)
-                return;
-
-            // Read each atlas found and its metadata
-            foreach (string path in atlasPaths)
+            if (atlasPaths != null && atlasPaths.Length != 0)
             {
-                // Get source atlas
-                Texture2D atlasTexture = LoadAtlasTextureFromPNG(path);
-                if (atlasTexture == null)
-                {
-                    Debug.LogWarningFormat("LoadSpellIconPacks(): Could not load spell icons atlas texture '{0}'", path);
-                    continue;
-                }
-
-                // Attempt to load metadata JSON file
-                SpellIconPack pack = null;
-                string packKey = Path.GetFileNameWithoutExtension(path);
-                string metadataFilename = packKey + ".txt";
-                string metadataPath = Path.Combine(sourcePath, metadataFilename);
-                if (File.Exists(metadataPath))
-                {
-                    // Try to load existing metadata and icons
-                    pack = ReadMetadata(metadataPath);
-                    LoadPackIcons(pack, atlasTexture, metadataPath);
-                }
-                else
+                // Read each atlas found and its metadata
+                foreach (string path in atlasPaths)
                 {
-                    // Create empty metadata file if none found
-                    pack = CreateIconPackEmptyMetadata(atlasTexture, metadataPath);
-                    return;
```

Comment:
I kept this return but i believe it shouldn't be there. Can you confirm what is the appropriate behaviour here?

Reply:
You can likely remove because safety check on following code won't do anything with 0 row count or icon count. But it's also a perfectly valid place to return as no further work needed if spell pack is empty.

Reply:
If meta data is missing for a pack, all the following packs are also skipped. Actually i think i found a bigger issue: a default file is created during startup scene, but then this one is loaded after transition to game scene, raising an exception due to bad operation `atlas.width / pack.rowCount` with zero rows.

This happens because DaggerfallUI.Start() is called again; Spell packs initialization should be moved to Awake or, even better, when requested the first time i think.

I also saw this when testing with two copies of test pack.
![iconPacks](https://user-images.githubusercontent.com/24359151/61080105-98bfdf00-a424-11e9-90d1-37a7972cb79d.png)

I can work on these issues with a new PR if you want.

Reply:
Of course, thank you for the help. :)